### PR TITLE
Move doc-values classes needed by ST_INTERSECTS to server

### PR DESCRIPTION
**Pull request by craigtaverner**
These classes are needed by ESQL spatial queries, and are not licensed in a way that prevents this move. Since they depend on lucene it is not possible to move them to a library. Instead they are moved to be co-located with the GeoPoint doc-values classes that already exist in server.

### CODE REVIEW

**Review by iverase — CHANGES_REQUESTED** [review]
Can we please move the under a package in ~~`org.apache.lucene`~~  `org.elasticsearch.lucene`? They are really implementing some lucene functionality.

File: x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/fielddata/Tile2DVisitorTests.java [hunk 1]
```diff
@@ -287,4 +290,40 @@ static void assertRelation(GeometryDocValueReader reader, Extent extent, GeoRela
         reader.visit(tile2DVisitor);
         assertThat(tile2DVisitor.relation(), in(expectedRelation));
     }
+
+    private static class TestCoordinateEncoder implements CoordinateEncoder {
```
Comment (iverase) [inline x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/fielddata/Tile2DVisitorTests.java:294]: nice!

**Review by iverase — APPROVED** [review]
lgtm

**Comment by craigtaverner** [comment]
@elasticsearchmachine run elasticsearch-ci/part-1

**Pull request merged**